src/subdomains/generic/user/models/user/user.service.ts

```diff
@@ -240,21 +245,44 @@ export class UserService {
     return this.userRepo.save({ ...user, ...update });
   }
 
-  async deactivateUser(id: number, allUser = false): Promise<void> {
-    const mainUser = await this.userRepo.findOne({
-      where: { id },
-      relations: { userData: { users: true, kycSteps: true } },
+  async updateAddress(userDataId: number, address: string, dto: UpdateAddressDto): Promise<UserV2Dto> {
+    const userData = await this.userDataRepo.findOne({
+      where: { id: userDataId },
+      relations: { users: { wallet: true } },
     });
-    if (!mainUser) throw new NotFoundException('User not found');
-    if (mainUser.userData.isBlockedOrDeactivated) throw new BadRequestException('User Account already deactivated');
-    if (mainUser.isBlockedOrDeactivated) throw new BadRequestException('User already deactivated');
+    if (!userData) throw new NotFoundException('User not found');
+
+    const userToUpdate = userData.users.find((u) => u.address === address);
+    if (!userToUpdate) throw new NotFoundException('Address not found');
+
+    userToUpdate.label = dto.label;
+    await this.userRepo.save(userToUpdate);
+
+    return UserDtoMapper.mapUser(userData);
+  }
 
-    if (!allUser) {
-      await this.userRepo.update(...mainUser.deactivateUser('Manual user deactivation'));
+  async deactivateUser(userDataId: number, address?: string): Promise<void> {
+    const userData = await this.userDataRepo.findOne({
+      where: { id: userDataId },
+      relations: { users: { wallet: true }, kycSteps: true },
+    });
+    if (!userData) throw new NotFoundException('User account not found');
+    if (userData.isBlockedOrDeactivated) throw new BadRequestException('User account already deactivated');
+
+    if (address) {
+      const user = userData.users.find((u) => u.address === address);
+      if (!user) throw new NotFoundException('Address not found');
+      if (user.isBlockedOrDeactivated) throw new BadRequestException('Address already deactivated');
+
+      await this.userRepo.update(...user.deactivateUser('Manual user deactivation'));
       return;
     }
 
-    await this.userDataService.deactivateUserData(mainUser.userData);
+    await this.userDataService.deactivateUserData(userData);
+
+    for (const user of userData.users) {
+      await this.userRepo.update(...user.deactivateUser('Manual user account deactivation'));
+    }
```

Comment:
Remove these lines here. We only deactivate the userData without the users from now on